docs: document log order (#5463)
vercel/turbo: docs/pages/repo/docs/reference/command-line-reference/run.mdx

- `{}` allows for a comma-separated list of "or" expressions
- `!` at the beginning of a pattern will negate the match

### `--log-order`

`type: string`

Set type of task output order. Defaults to "auto" to let

> turbo decide based on its own heuristics.

Reviewer: We're mentioning "using heuristics" but I'm left wondering what _are_ those heuristics? What behaviors should I be expecting to see?

| option  | description                               |
| ------- | ----------------------------------------- |
| auto    | turbo decides based on its own heuristics |

> | stream  | show output as soon as it is available    |

Reviewer: Start phrase with a capital letter.

> | grouped | show output grouped by task               |

Reviewer: Capital letter as well!

**Example**

```shell
turbo run build --log-order=stream
turbo run build --log-order=grouped
```

<Callout type="info">

>   If log order is set to auto and turbo detects that it is running on Github Actions, then turbo will

Reviewer: turbo should be code formatted. `turbo`

  create [grouped logs](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#grouping-log-lines).

>   If this isn't desired, then the behavior can be opted out of by setting the log order to stream or grouped directly.

Reviewer: ```suggestion
  You can opt out of this behavior by setting a log order of your own.
```
